### PR TITLE
feat(README.md): update Node.js version requirement to v22.14 or high…

### PR DESCRIPTION
**Pull request by fra-intini**
…er for compatibility

fix(README.md): correct file names in the setup section to use hyphens instead of camel case
feat(tests): add comprehensive test cases for abbreviation handling, constructor, edge cases, language support, performance, and private methods to ensure tokenizer functionality
feat(utils): implement functions to download NLTK data, convert it to JSON, and manage language mappings for improved data handling
chore(package.json): update scripts for testing and add post-install script for automatic data setup
fix(punkt.ts): adjust path imports to use the new utils directory structure for better organization
chore(jest.config.js): configure Jest for testing with module name mapping and specify test match patterns
chore: remove unused Italian test file to clean up the codebase

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          node-version: 22
+
+      - name: Install dependencies
+        run: npm i
+
+      - name: Run tests
+        run: npm run test
+
+      - name: Test coverage
```
Comment (guidev) [inline .github/workflows/test.yml:29]: che plugin è?

File: package.json [hunk 2]
```diff
@@ -8,29 +8,32 @@
...
+    "postinstall": "node dist/post-install.js",
     "dev": "ts-node src/index.ts",
     "build": "tsc",
     "prestart": "npm run build",
     "start": "node dist/index.js",
     "clean": "rm -rf dist",
     "prebuild": "npm run clean",
-    "test": "ts-node src/tests/itaTest.ts",
-    "test_builded": "node dist/tests/itaTest.js"
+    "test_builded": "node dist/tests/itaTest.js",
```
Comment (guidev) [inline package.json:18]: usiamo https://www.npmjs.com/package/ts-jest

**Review by guidev — CHANGES_REQUESTED** [review]

File: .github/workflows/dependabot-build.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22'
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build TypeScript
+        run: npm run build
```
[the remark this answers is not in the document]
Comment (fra-intini) [reply .github/workflows/dependabot-build.yml:28]: va bene, errore mio che non ricordavo che pushamo anche il buildato.

File: .github/workflows/test.yml (hunk 1 above)
Comment (fra-intini) [reply .github/workflows/test.yml:29]: È una action che crea un commit message con la tabella della test coverage. Se la si vuole rimuovere possiamo eliminare lo step
